mailbox/backup/src/main/java/org/apache/james/mailbox/backup/zip/ZipEntryType.java

```diff
@@ -0,0 +1,57 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.backup.zip;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public enum ZipEntryType {
+
+    MAILBOX(ZipEntryType.MAILBOX_VALUE),
+    MAILBOX_ANNOTATION_DIR(ZipEntryType.MAILBOX_ANNOTATION_DIR_VALUE),
+    MAILBOX_ANNOTATION(ZipEntryType.MAILBOX_ANNOTATION_VALUE),
+    MESSAGE(ZipEntryType.MESSAGE_VALUE);
+
+    final long value;
+
+    ZipEntryType(long value) {
+        this.value = value;
+    }
+
+    private static final long MAILBOX_VALUE = 1L;
```

Comment:
What is the purpose of these longs?

Reply:
To serialize the entry type has Long without depending on the ordinal.

Reply:
Ok so if you want that information to be stable over time, you should create an archive (in some format that won't be easily refactorable) that uses these values and check that restore is producing the right result.

Reply:
Ce n'est pas un peu overkill ? Si l'on fait  ça, on devrait aussi le faire pour tout les ZipHeader id des extraFields, et pour la serialization des Flags.
Qu'est ce que tu en  penses si je renomme le value en serializationValue ? Ça serait plus clair et les developpeurs futures comprendront qu'il ne faut pas changer les valeurs ?

Reply:
Isn't it a little too much. If we do that we should do the same thing for the ZipHeader id of all the extra fields, and for the flags serialization.
Would you agree if i simply rename 'value' to 'serializedValue' ? The usage and expectation of stability of this value would be more explicit.

Reply:
I will add a unit test to enforce that the values do not change, like it's done with the header for the other extra fields